nomenclature/testing.py

```diff
@@ -2,10 +2,20 @@
 import logging
 from pathlib import Path
 
+import click
+
+import nomenclature
 
 logger = logging.getLogger(__name__)
 
 
+@click.group(chain=True, invoke_without_command=True)
+@click.argument("path", type=click.Path(exists=True))
```

Comment:
Nice!

Reply:
Thinking about it now, this makes sense to integrate if we then could change `Path(path)` to `path` in the rest of the code, but I guess since the functions should run also without the CLI we can't reduce it to `path`. But I guess also no harm in implementing it without changing the rest. Or am I missing something?

Reply:
Ah very good point, I missed that. It of course also has to work outside of the CLI. Seems like we'll have to convert to path after all. Maybe with a check to see if `path` is already a `Path` object. Something like this:
```python
if not isinstance(path, Path) and isinstance(path, str):
    path = Path(path)
elif not isinstance(path, Path):
    raise TypeError("path must be Union[str, pathlib.Path]")
```

Alternatively there might be a shortcut using pydantic (yes, I know, I'm recommending it again). The [`validate_arguments`](https://pydantic-docs.helpmanual.io/usage/validation_decorator/) decorator can be used to validate data types for functions. That might be worth checking out as it could make our lives easier, also for other applications

Reply:
:D not at all.
If we implement it like that I have the feeling it could turn out a bit over engineered for the current use case. Or maybe I'm mistaken and the use case at this state is not obvious to me or it's better to just implement it now for future applications. Right now I would vote for now going with `@click.argument("path", type=click.Path(exists=True, file_okay=False, dir_okay=True))` and leave the conversion. What do you think?

Reply:
Yeah I think we can get away with leaving the type out for non CLI use. Maybe just to help the user but keep the overhead for you minimal you could add type hints to `cli(path: Path)`, `assert_valid_yaml...` and `assert_valid_structure...` to indicate to the user that we in fact want a `pathlib.Path`.

Reply:
Perfect, already did 👍 We do it like that.

Reply:
Perfect